src/main/java/billgates/entities/UserFactory.java

```diff
@@ -0,0 +1,30 @@
+package billgates.entities;
+
+public class UserFactory {
+    //use getShape method to get object of type shape
+    public IUser getUser(int id, String name, String UserType) {
+        if (UserType == "DummyUser") {
+            IUser d = new DummyUser(id, name);
+            return d;
+        }
+        if (UserType == "User") {
+            IUser user = User.getInstance(id, name, "TemporyPW", 0);
+
+            return user;
+        }
+        return null;
```

Comment:
An error raise could be more appropriate here..?

Reply:
That is correct, but as I have mentioned earlier, if we will stick to the factory design pattern, we shouldn't have too much places to raise an error in the UserFactory class.